### PR TITLE
Phase management

### PR DESCRIPTION
**Pull request by rzulian**
This is introducing a proper phase management. 
The standard workflow is:
SetupInit -> SetupFaction -> SetupBuilding -> SetupBooster -> RoundStart -> RoundIncome ->RoundGaia -> RoundMove -> RoundFinish -> endGame

each phase has a phaseBegin and a onPhaseEnd.
code can terminate a phase with `phaseEnd()`.
the standard  workflow is checking for `turnOrder.length === 0` to call phaseEnd.

What's missing:

- [ ] move LeechPhase to a standard phase

- [ ] move Mine after temp booster as a standard phase

### CODE REVIEW

File: src/player-data.ts [hunk 1]
```diff
@@ -215,21 +215,21 @@ export default class PlayerData extends EventEmitter {
...
 
-  discardGaiaPower(power: number){
-    if ( this.brainstone === BrainstoneArea.Gaia){
-      this.brainstone = BrainstoneArea.Transit;
-      power -= 1;
-    }
+  returnGaiaPower(power: number) {
+    // this is used only by Terrans for the conversion
+    // they are pay tg to get x
+    this.power.area1 += this.hasPlanetaryInstitute() ? 0 : power;
```
Comment (coyotte508) [inline src/player-data.ts]: This function is only called if Terrans have their planetary institute. Terrans always get their power tokens in area 2, even if they have no planetary institute.

File: src/player.ts [hunk 2]
```diff
@@ -385,6 +397,7 @@ export default class Player extends EventEmitter {
   gaiaPhase() {
     /* Move gaia power tokens to regular power areas */
     // Terrans move directly to power area 2
+    // for Terrans is done  after gaia Income selection
     if (this.faction === Faction.Terrans) {
       this.data.power.area2 += this.data.power.gaia;
       if (this.data.brainstone === BrainstoneArea.Gaia ) {
```
Comment (coyotte508) [inline src/player.ts]: This test is always false, terrans have no brain stone.

File: src/player-data.ts (hunk 1 above)
Comment (rzulian) [reply src/player-data.ts]: oh right, sorry

File: src/player.ts (hunk 2 above)
Comment (rzulian) [reply src/player.ts]: yes, I thought I already corrected this ...

**Pull request merged**